From 7c31d5c2398c9ca5d1b15ddee82272a660eab49e Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Wed, 21 Oct 2015 09:51:26 +0800
Subject: [PATCH] MLK-11745 rtc: rtc-snvs: keep snvs clock before poweroff

Keep SNVS clock before poweroff by SW, otherwise PMIC_ON_REQ can't be pulled
up if ONOFF key pressed next time. This is design limitation.

Signed-off-by: Robin Gong <b38343@freescale.com>
---
 drivers/rtc/rtc-snvs.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-snvs.c b/drivers/rtc/rtc-snvs.c
index cb19c83..ede0750 100644
--- a/drivers/rtc/rtc-snvs.c
+++ b/drivers/rtc/rtc-snvs.c
@@ -267,7 +267,11 @@ static void snvs_poweroff(void)
 	value = readl(snvs_base + SNVS_LPCR);
 	/* set TOP and DP_EN bit */
 	writel(value | 0x60, snvs_base + SNVS_LPCR);
-	clk_disable(clk_snvs);
+	/*
+	 * Do not turn off snvs clock otherwise PMIC_ON_REQ can't be pulled
+	 * high by press ONOFF. This is design limitation.
+	 */
+	/* clk_disable(clk_snvs); */
 }
 
 static int snvs_rtc_probe(struct platform_device *pdev)
-- 
1.8.0

